to_copy/usr/local/unbound-1.7/sbin/filter_client.py

```diff
@@ -46,6 +56,17 @@ def operate(id, event, qstate, qdata):
     if (event == MODULE_EVENT_NEW) or (event == MODULE_EVENT_PASS):
         name = qstate.qinfo.qname_str.rstrip('.')
 
+        if not is_blacklist_empty():
+            if name in doh_canary_domains:
+                qstate.return_rcode = RCODE_NXDOMAIN
+                qstate.ext_state[id] = MODULE_FINISHED
+                return True            
+
+            if name in doh_provider_domains:
```

Comment:
We should probably check if it's in the provider domains *and* not in the blacklist for the case of a PEP admin wants to make those lookups fail if they really want. So... `if name in doh_provider_domains and not is_blocked(name):` ? Or alternatively (if this works right):
```
if self.disable_doh and name in doh_canary_domains:
   qstate.return__rcode = RCODE_NXDOMAIN
   qstate.ext_state[id] = MODULE_FINISHED
   return True
```

Saves a send_request per lookup if i'm reading this correctly. Could be wrong.

Reply:
Changes were made to send just one request and checks if it is in the provider list and not in the blacklist.